src/set_theory/ordinal/arithmetic.lean

```diff
@@ -2178,17 +2181,14 @@ le_iff_le_iff_lt_iff_lt.2 nat_lt_card
 by simp only [le_antisymm_iff, card_le_nat, nat_le_card]
 
 @[simp] theorem type_fin (n : ℕ) : @type (fin n) (<) _ = n :=
-by rw [← card_eq_nat, card_type, mk_fin]
-
-@[simp] theorem lift_nat_cast (n : ℕ) : lift n = n :=
-by induction n with n ih; [simp only [nat.cast_zero, lift_zero],
-  simp only [nat.cast_succ, lift_add, ih, lift_one]]
+by rw [←card_eq_nat, card_type, mk_fin]
 
-theorem lift_type_fin (n : ℕ) : lift (@type (fin n) (<) _) = n :=
-by simp only [type_fin, lift_nat_cast]
+@[simp] theorem lift_nat_cast : ∀ n : ℕ, lift.{u v} n = n
```

Comment:
Do we have `lift` as a `ring_hom`? If yes, then we can use the global `map_nat_cast`.

Reply:
We don't currently have this, but I'll make this a followup PR.

Reply:
Oh actually, we can't have this, since ordinals aren't a `non_assoc_semiring` (non-commutative addition).